rpc/modules: TestStateModule_GetPairs_All fails sometimes

<!---

PLEASE READ CAREFULLY

-->

## Expected Behavior

`TestStateModule_GetPairs_All` should pass at all times.


-

## Current Behavior

```
t=2020-05-15T17:28:03+0200 lvl=trce msg=[babe] authority index=0
--- FAIL: TestStateModule_GetPairs_All (0.31s)
    state_test.go:60: 
                Error Trace:    state_test.go:60
                Error:          Not equal: 
                                expected: []interface {}{[]string{"0x3a6b657931", "0x76616c756531"}, []string{"0x3a6b657932", "0x76616c756532"}}
                                actual  : []interface {}{[]string{"0x3a6b657932", "0x76616c756532"}, []string{"0x3a6b657931", "0x76616c756531"}}
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,6 @@
                                 ([]interface {}) (len=2) {
                                + ([]string) (len=2) {
                                +  (string) (len=12) "0x3a6b657932",
                                +  (string) (len=14) "0x76616c756532"
                                + },
                                  ([]string) (len=2) {
                                @@ -4,6 +8,2 @@
                                   (string) (len=14) "0x76616c756531"
                                - },
                                - ([]string) (len=2) {
                                -  (string) (len=12) "0x3a6b657932",
                                -  (string) (len=14) "0x76616c756532"
                                  }
                Test:           TestStateModule_GetPairs_All
```

-

## Possible Solution

<!---

Not obligatory, but this is the place to suggest the underlying cause and
possible fix for the bug, if you have one, or ideas on how to implement the
fix. We'll be sure to credit your ideas in the commit log, or better yet,
submit a PR and you'll get credit for the whole thing.

-->

-

## Steps to Reproduce

<!---

This is the most important information you can give us for a bug report.
Without good information, it will take much longer to resolve your issue!

The best strategy here is to assume the maintainer reading this just started
working on the project yesterday.

If possible, please provide a link to a live example, or an unambiguous set of
steps to reproduce this bug. Include code to reproduce, if relevant.

-->

1. run `TestStateModule_GetPairs_All` multiple times until it fails
2.
3.
4.

## Specification

<!---

Example specification (feel free to copy and paste if applicable or delete the
specification section if a specification is not applicable):

- go version: `1.13.8`
- gossamer version: `development`
- gossamer commit tag: NA
- gossamer commit hash: NA
- operating system: Ubuntu 19.10
- additional links: NA

-->

- go version:
- gossamer version:
- gossamer commit tag:
- gossamer commit hash:
- operating system:
- additional links:

## Checklist

<!---

Each empty square brackets below is a checkbox. Replace [ ] with [x] to check
the box after completing the task.

--->

- [ ] I have read [CONTRIBUTING](CONTRIBUTING.md) and [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)
- [ ] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

<!--- Modified from trufflesuite/ganache -->

--- a/dot/rpc/modules/state_test.go
+++ b/dot/rpc/modules/state_test.go
@@ -17,6 +17,7 @@ package modules
 
 import (
 	"fmt"
+	"sort"
 	"testing"
 
 	"github.com/ChainSafe/gossamer/lib/common"
@@ -57,6 +58,9 @@ func TestStateModule_GetPairs_All(t *testing.T) {
 
 	err := sm.GetPairs(nil, &req, &res)
 	require.NoError(t, err)
+	sort.Slice(res, func(i, j int) bool {
+		return res[i].([]string)[0] < res[j].([]string)[0]
+	})
 	require.Equal(t, expected, res)
 
 }

--- a/lib/runtime/helpers_test.go
+++ b/lib/runtime/helpers_test.go
@@ -19,6 +19,7 @@ import (
 )
 
 var kr, _ = keystore.NewSr25519Keyring()
+var maxRetries = 10
 
 func TestExportRuntime(t *testing.T) {
 	fp := "runtime.out"
@@ -266,14 +267,16 @@ func TestFinalizeBlock(t *testing.T) {
 	}
 
 	err := rt.InitializeBlock(header)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 
-	res, err := rt.FinalizeBlock()
-	if err != nil {
-		t.Fatal(err)
+	var res *types.Header
+	for i := 0; i < maxRetries; i++ {
+		res, err = rt.FinalizeBlock()
+		if err == nil {
+			break
+		}
 	}
+	require.NoError(t, err)
 
 	res.Number = header.Number
 
@@ -368,7 +371,12 @@ func TestApplyExtrinsic_StorageChange_Set(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, []byte("testvalue"), val)
 
-	_, err = rt.FinalizeBlock()
+	for i := 0; i < maxRetries; i++ {
+		_, err = rt.FinalizeBlock()
+		if err == nil {
+			break
+		}
+	}
 	require.NoError(t, err)
 
 	val, err = rt.storage.GetStorage([]byte("testkey"))

